ismatec/util.py

```diff
@@ -14,21 +18,22 @@
 
 
 class Communicator(threading.Thread):
```

Comment:
The communicator has methods for caching booleans related to pump state. This is not the right home for it.

Communicator/Transport should just be about how to send data over the wire to the device. The boolean caching feels like it belongs in the driver, and will make for more readable code if it lives there.

Haven't tried it, just a hunch.

Reply:
Agreed it's ugly.  Someone needs catch the "hi I'm a pump and I am running/just stopped" commands that are sent without being requested.  I couldn't think of a way for the driver layer to catch these.

Reply:
Moved to `Pump` with a callback in
https://github.com/numat/ismatec/commit/fd98ee74cbf9de71af28a5c21c4c80bed1d5f865